### DISCUSSION TITLE
Do not deprecate task_allow_error_cb_on_chord_header

### DISCUSSION THREAD

**Discussion by jerr0328** [celery/celery#8318] [General]
In Celery 5.3, a new warning appears with chords and the link_error callback due to #7712 :

```
/opt/venv/lib/python3.11/site-packages/celery/canvas.py:2286: CPendingDeprecationWarning: task_allow_error_cb_on_chord_header=False is pending deprecation in a future release of Celery.
Please test the new behavior by setting task_allow_error_cb_on_chord_header to True and report any concerns you might have in our issue tracker before we make a final decision regarding how errbacks should behave when used with chords.
```

I think both options are valid, but in the situation where we're using chords and errbacks, the current behavior (where task_allow_error_cb_on_chord_header=False) is still desired. We use the errback in those cases to unlock a lock on Redis that keeps a pipeline from being executed in parallel (we want other tasks to be executed in parallel, but all the ones from this pipeline should be in run in order and without other instances of the pipeline running at the same time). If the callback is called for each failing header task + the body task of the chord, it will mean we need to handle the callback being called multiple times, rather than a single time for the failure. While we could workaround if the flag is removed and the behavior is changed in Celery 6.0, I can't imagine this being the only case where the current logic would be preferred indefinitely.

I didn't want to make this a bug report as I know the warning was intentional, but I figured a Discussion is probably a better place for such a discussion to get more input from others in the Celery community.

**Comment by auvipy** [comment]
@celery/core-developers we got an interesting discussion here I guess!

**Comment by Nusnus** [comment]
Thank you for this discussion @jerr0328 !
Indeed, this isn't a bug but a change of behavior, which, as the warning said, requires discussion; so I am very glad to see this conversation!

To the subject,

The new feature flag allows changing this behavior and indeed warns about plans to remove it completely, breaking the API 100%. That being said, it was not set in stone, which is why we've asked the community for feedback.

I think your point makes sense _against_ general deprecation, and I support your argument, but I am still not sure about the default value.

## False
- Chord's error handling remains on body only.
- Default behavior remains as before / doesn't break compatibility.
## True
- Chord's error handling is called for each failing header **AND** for the body (that is **one** errback called **for each** failed header task + **another** errback call for the body).
- Breaking API (_might_ require refactoring the error handlers to be idempotent, for example)

---
Lastly, for now, and until further notice, it will remain as-is, meaning the flag **will not** be removed, and the behavior **will stay** the _Falsy_ way. This change has a tremendous effect, and we are not in a hurry to apply the _Truty_ flow without confidence it will serve well the majority of the community _at least_, and as far as we can understand from additional feedback from the community following this thread/discussion.
